src/tracing/types.rs

```diff
@@ -353,31 +373,44 @@ impl CallTraceNode {
                 to: self.trace.address,
                 value: self.trace.value,
                 gas: self.trace.gas_limit,
-                input: self.trace.data.clone(),
+                input: Bytes::new(),
+                // input: self.trace.data.clone(),
                 call_type: self.kind().into(),
                 // tx_type: self.trace.tx_type,
             }),
             CallKind::Create | CallKind::Create2 => Action::Create(CreateAction {
                 from: self.trace.caller,
                 value: self.trace.value,
                 gas: self.trace.gas_limit,
-                init: self.trace.data.clone(),
+                init: Bytes::new(),
+                // init: self.trace.data.clone(),
                 creation_method: self.kind().into(),
             }),
         }
     }
 
     /// Converts this call trace into an _empty_ geth [CallFrame]
     pub fn geth_empty_call_frame(&self, include_logs: bool) -> CallFrame {
+        self.geth_empty_call_frame_with_masking(include_logs, false)
+    }
+
+    /// Converts this call trace into an _empty_ geth [CallFrame] with optional data masking
+    pub fn geth_empty_call_frame_with_masking(&self, include_logs: bool, is_root_call: bool) -> CallFrame {
         let mut call_frame = CallFrame {
             typ: self.trace.kind.to_string(),
             from: self.trace.caller,
             to: Some(self.trace.address),
             value: Some(self.trace.value),
             gas: U256::from(self.trace.gas_limit),
             gas_used: U256::from(self.trace.gas_used),
-            input: self.trace.data.clone(),
-            output: (!self.trace.output.is_empty()).then(|| self.trace.output.clone()),
+            // Mask input for nested calls
+            input: Bytes::new(),
+            // Mask output for nested calls (keep for root call)
+            output: if is_root_call {
```

Comment:
dont have to change but for future, in rust, i think its way better to use match statements instead of if/elses. reads better imo